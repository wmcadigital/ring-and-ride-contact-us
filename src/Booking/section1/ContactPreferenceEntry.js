import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import ProgressIndicator from "../../common/ProgressIndicator";
import Question from "../../common/Question";
import CheckboxContainer from "../../common/CheckboxContainer";
import Checkbox from "../../common/Checkbox";
import TextInput from "../../common/TextInput";
import FieldError from "../../common/FieldError";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutSectionName from "./getAboutSectionName";

const ContactPreferenceEntry = () => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const question =
    formValues["bookingParty"] === "behalfSomeone"
      ? `How would ${formValues["firstName"]} like to be contacted?`
      : "How would you like to be contacted?";

  const posssesiveDeterminer =
    formValues["bookingParty"] === "behalfSomeone" ? "their" : "your";

  const errorContactPrefReqd = stateApi.submitFailed
    ? stateApi.errors?.contactPreference
    : null;

  const errorEmailAddress = stateApi.submitFailed
    ? stateApi.errors?.emailAddress
    : null;

  const errorPhoneNo = stateApi.submitFailed ? stateApi.errors?.phoneNo : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("1", stateApi)}
        sectionName={getAboutSectionName(stateApi)}
      />
      <Question text={question} />
      <CheckboxContainer
        error={errorContactPrefReqd}
        description={`We'll use this to get in touch about ${posssesiveDeterminer} booking.`}
      >
        <FieldError text={errorContactPrefReqd} />
        <Checkbox fieldName="emailContact" label="Email" />
        {formValues.emailContact ? (
          <div className="wmrards-m-l-xl wmrards-p-l-sm">
            <TextInput
              fieldName="emailAddress"
              label={
                <>
                  Email address
                  <p>For example, name@example.com</p>
                </>
              }
              error={errorEmailAddress}
            />
          </div>
        ) : null}
        <Checkbox fieldName="phoneContact" label="Phone" />
        {formValues.phoneContact ? (
          <div className="wmrards-m-l-xl wmrards-p-l-sm">
            <TextInput
              fieldName="phoneNo"
              label={
                <>
                  Phone number
                  <p>For example, 07700900457</p>
                </>
              }
              error={errorPhoneNo}
            />
          </div>
        ) : null}
      </CheckboxContainer>
    </FormSection>
  );
};

export default ContactPreferenceEntry;
