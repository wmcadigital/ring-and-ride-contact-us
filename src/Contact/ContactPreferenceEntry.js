import { useFormState } from "react-final-form";

import ProgressIndicator from "../common/ProgressIndicator";
import Question from "../common/Question";
import Checkbox from "../common/Checkbox";
import TextInput from "../common/TextInput";
import FieldError from "../common/FieldError";

const ContactPreferenceEntry = () => {
  const stateApi = useFormState();
  const formValues = stateApi.values;

  const errorContactPrefReqd = stateApi.submitFailed
    ? stateApi.errors?.contactPreference
    : null;

  const errorEmailAddress = stateApi.submitFailed
    ? stateApi.errors?.emailAddress
    : null;

  const errorPhoneNo = stateApi.submitFailed ? stateApi.errors?.phoneNo : null;

  return (
    <div className="wmrards-fe-group wmrards-m-t-20">
      <ProgressIndicator
        sectionPosition="Section 2 of 2"
        sectionName="About you"
      />
      <Question text="How would you like to be contacted?" />
      <div className="wmrards-fe-group">
        <div
          className={`wmrards-fe-checkboxes ${
            errorContactPrefReqd && "wmrards-fe-group--error"
          }`}
        >
          <div className="wmrards-fe-checkboxes__desc">
            We&apos;ll use this to get in touch about your enquiry.
          </div>
          <FieldError text={errorContactPrefReqd} />
          <Checkbox fieldName="emailContact" label="Email" />
          {formValues.emailContact ? (
            <div className="wmrards-m-l-xl wmrards-p-l-sm">
              <TextInput
                fieldName="emailAddress"
                label={
                  <>
                    Email address
                    <p>For example, name@example.com</p>
                  </>
                }
                error={errorEmailAddress}
              />
            </div>
          ) : null}
          <Checkbox fieldName="phoneContact" label="Phone" />
          {formValues.phoneContact ? (
            <div className="wmrards-m-l-xl wmrards-p-l-sm">
              <TextInput
                fieldName="phoneNo"
                label={
                  <>
                    Phone number
                    <p>For example, 07700900457</p>
                  </>
                }
                error={errorPhoneNo}
              />
            </div>
          ) : null}
        </div>
      </div>
    </div>
  );
};

export default ContactPreferenceEntry;
