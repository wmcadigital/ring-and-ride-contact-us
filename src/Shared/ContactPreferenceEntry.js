import { useFormState } from "react-final-form";
import PropTypes from "prop-types";

import FormSection from "../common/FormSection";
import ProgressIndicator from "../common/ProgressIndicator";
import Question from "../common/Question";
import CheckboxContainer from "../common/CheckboxContainer";
import Checkbox from "../common/Checkbox";
import TextInput from "../common/TextInput";
import FieldError from "../common/FieldError";

const ContactPreferenceEntry = ({description}) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;

  const errorContactPrefReqd = stateApi.submitFailed
    ? stateApi.errors?.contactPreference
    : null;

  const errorEmailAddress = stateApi.submitFailed
    ? stateApi.errors?.emailAddress
    : null;

  const errorPhoneNo = stateApi.submitFailed ? stateApi.errors?.phoneNo : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition="Section 2 of 2"
        sectionName="About you"
      />
      <Question text="How would you like to be contacted?" />
      <CheckboxContainer
        error={errorContactPrefReqd}
        description={description}
      >
        <FieldError text={errorContactPrefReqd} />
        <Checkbox fieldName="emailContact" label="Email" />
        {formValues.emailContact ? (
          <div className="wmrards-m-l-xl wmrards-p-l-sm">
            <TextInput
              fieldName="emailAddress"
              label={
                <>
                  Email address
                  <p>For example, name@example.com</p>
                </>
              }
              error={errorEmailAddress}
            />
          </div>
        ) : null}
        <Checkbox fieldName="phoneContact" label="Phone" />
        {formValues.phoneContact ? (
          <div className="wmrards-m-l-xl wmrards-p-l-sm">
            <TextInput
              fieldName="phoneNo"
              label={
                <>
                  Phone number
                  <p>For example, 07700900457</p>
                </>
              }
              error={errorPhoneNo}
            />
          </div>
        ) : null}
      </CheckboxContainer>
    </FormSection>
  );
};

ContactPreferenceEntry.propTypes = {
  description: PropTypes.string,
};

export default ContactPreferenceEntry;
