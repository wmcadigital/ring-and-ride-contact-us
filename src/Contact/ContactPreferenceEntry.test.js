import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import ContactPreferenceEntry from "./ContactPreferenceEntry";

jest.mock("react-final-form");

const formApiReturn = {
  values: {
    emailContact: true,
    emailAddress: "test@testerson.com",
    phoneContact: true,
    phoneNo: "0900 555 8888",
  },
};

describe("ContactPreferenceEntry", () => {
  it("renders ContactPreferenceEntry (no values)", () => {
    useFormState.mockImplementationOnce(() => ({ values: {} }));
    const renderer = createRenderer();

    renderer.render(<ContactPreferenceEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders ContactPreferenceEntry (all values)", () => {
    useFormState.mockImplementationOnce(() => formApiReturn);
    const renderer = createRenderer();

    renderer.render(<ContactPreferenceEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders ContactPreferenceEntry with error if present (no values)", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {},
      submitFailed: true,
      errors: { contactPreference: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<ContactPreferenceEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders ContactPreferenceEntry with error if present (all values)", () => {
    useFormState.mockImplementationOnce(() => ({
      values: { ...formApiReturn.values, phoneNo: "abc" },
      submitFailed: true,
      errors: { phoneNo: "Invalid" },
    }));
    const renderer = createRenderer();

    renderer.render(<ContactPreferenceEntry />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
