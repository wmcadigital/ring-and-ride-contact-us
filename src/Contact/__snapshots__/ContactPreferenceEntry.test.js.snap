// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactPreferenceEntry renders ContactPreferenceEntry (all values) 1`] = `
<div
  className="wmrards-fe-group wmrards-m-t-20"
>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="How would you like to be contacted?"
  />
  <div
    className="wmrards-fe-group"
  >
    <div
      className="wmrards-fe-checkboxes null"
    >
      <div
        className="wmrards-fe-checkboxes__desc"
      >
        We'll use this to get in touch about your enquiry.
      </div>
      <FieldError
        text={null}
      />
      <Checkbox
        fieldName="emailContact"
        label="Email"
      />
      <div
        className="wmrards-m-l-xl wmrards-p-l-sm"
      >
        <TextInput
          error={null}
          fieldName="emailAddress"
          label={
            <React.Fragment>
              Email address
              <p>
                For example, name@example.com
              </p>
            </React.Fragment>
          }
        />
      </div>
      <Checkbox
        fieldName="phoneContact"
        label="Phone"
      />
      <div
        className="wmrards-m-l-xl wmrards-p-l-sm"
      >
        <TextInput
          error={null}
          fieldName="phoneNo"
          label={
            <React.Fragment>
              Phone number
              <p>
                For example, 07700900457
              </p>
            </React.Fragment>
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ContactPreferenceEntry renders ContactPreferenceEntry (no values) 1`] = `
<div
  className="wmrards-fe-group wmrards-m-t-20"
>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="How would you like to be contacted?"
  />
  <div
    className="wmrards-fe-group"
  >
    <div
      className="wmrards-fe-checkboxes null"
    >
      <div
        className="wmrards-fe-checkboxes__desc"
      >
        We'll use this to get in touch about your enquiry.
      </div>
      <FieldError
        text={null}
      />
      <Checkbox
        fieldName="emailContact"
        label="Email"
      />
      <Checkbox
        fieldName="phoneContact"
        label="Phone"
      />
    </div>
  </div>
</div>
`;

exports[`ContactPreferenceEntry renders ContactPreferenceEntry with error if present (all values) 1`] = `
<div
  className="wmrards-fe-group wmrards-m-t-20"
>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="How would you like to be contacted?"
  />
  <div
    className="wmrards-fe-group"
  >
    <div
      className="wmrards-fe-checkboxes undefined"
    >
      <div
        className="wmrards-fe-checkboxes__desc"
      >
        We'll use this to get in touch about your enquiry.
      </div>
      <FieldError
        text={null}
      />
      <Checkbox
        fieldName="emailContact"
        label="Email"
      />
      <div
        className="wmrards-m-l-xl wmrards-p-l-sm"
      >
        <TextInput
          fieldName="emailAddress"
          label={
            <React.Fragment>
              Email address
              <p>
                For example, name@example.com
              </p>
            </React.Fragment>
          }
        />
      </div>
      <Checkbox
        fieldName="phoneContact"
        label="Phone"
      />
      <div
        className="wmrards-m-l-xl wmrards-p-l-sm"
      >
        <TextInput
          error="Invalid"
          fieldName="phoneNo"
          label={
            <React.Fragment>
              Phone number
              <p>
                For example, 07700900457
              </p>
            </React.Fragment>
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ContactPreferenceEntry renders ContactPreferenceEntry with error if present (no values) 1`] = `
<div
  className="wmrards-fe-group wmrards-m-t-20"
>
  <ProgressIndicator
    sectionName="About you"
    sectionPosition="Section 2 of 2"
  />
  <Question
    text="How would you like to be contacted?"
  />
  <div
    className="wmrards-fe-group"
  >
    <div
      className="wmrards-fe-checkboxes wmrards-fe-group--error"
    >
      <div
        className="wmrards-fe-checkboxes__desc"
      >
        We'll use this to get in touch about your enquiry.
      </div>
      <FieldError
        text="Required"
      />
      <Checkbox
        fieldName="emailContact"
        label="Email"
      />
      <Checkbox
        fieldName="phoneContact"
        label="Phone"
      />
    </div>
  </div>
</div>
`;
